src/hooks/useModalState.js

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+
+const ModalTypes = {
+  ADD: 'add',
+  EDIT: 'edit',
+  CONFIRM: 'confirm'
+};
+
+const INIT_STATE = { type: null, data: null };
+
+const useModalState = () => {
+  const [modal, setModal] = useState(INIT_STATE);
+
+  const isAddType = modal.type === ModalTypes.ADD;
+  const isEditType = modal.type === ModalTypes.EDIT;
+  const isConfirmType = modal.type === ModalTypes.CONFIRM;
+
+  const close = () => setModal(INIT_STATE);
+  const showAdd = () => setModal({ type: ModalTypes.ADD });
+  const showEdit = (data) => setModal({ type: ModalTypes.EDIT, data });
+  const showConfirm = (data) => setModal({ type: ModalTypes.CONFIRM, data });
+
+  const data = modal.data;
```

Comment:
With ESlint you would be getting warnings because you have this variable and shadowing it in 20-21 line... 
Please either rename this variable, or skip creating it and just
```
return {
  data: modal.data,
...
```